From 0dd14b7acf36b8ae1622c81713a3218a48349034 Mon Sep 17 00:00:00 2001
From: Rob Evers <revers@redhat.com>
Date: Mon, 23 Jun 2014 21:41:47 -0400
Subject: [scsi] lpfc 8.3.45: Fixed missing initialization for task management IOCBs

Message-id: <1403559726-20488-6-git-send-email-revers@redhat.com>
Patchwork-id: 84425
O-Subject: [RHEL7.1 e-stor PATCH 05/24] scsi: lpfc 8.3.45: Fixed missing initialization for task management IOCBs
Bugzilla: 1088574
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: James Smart <james.smart@emulex.com>

Signed-off-by: James Smart <james.smart@emulex.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit f9226c20b736916552d623ef632b7152f61dfcc7)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/lpfc/lpfc_scsi.c b/drivers/scsi/lpfc/lpfc_scsi.c
index 92f90df..462453e 100644
--- a/drivers/scsi/lpfc/lpfc_scsi.c
+++ b/drivers/scsi/lpfc/lpfc_scsi.c
@@ -4423,6 +4423,8 @@ lpfc_scsi_prep_task_mgmt_cmd(struct lpfc_vport *vport,
  }
  piocb->ulpFCP2Rcvy = (ndlp->nlp_fcp_info & NLP_FCP_2_DEVICE) ? 1 : 0;
  piocb->ulpClass = (ndlp->nlp_fcp_info & 0x0f);
+ piocb->ulpPU = 0;
+ piocb->un.fcpi.fcpi_parm = 0;
 
  /* ulpTimeout is only one byte */
  if (lpfc_cmd->timeout > 0xff) {
-- 
1.7.1